### ISSUE TITLE
Generate Excel Sheet is not working

### ISSUE THREAD

**Issue by lajish** [aspiresoftware/redmine-import-tasks#2]
Hi,

After installing this plugin, when I try to export Excel template, it is not working.

**Comment by guruprasad1** [comment]
Can you specify any Errors that get generated while exporting template?

**Comment by vnazarenko** [comment]
After installing plugin i try to configure it, and see this error 
http://monosnap.com/image/QOU0UEQkAeDbrRZ6SFyv3AwZJ8eJcF

**Comment by vnazarenko** [comment]
Ok got it, it's important that plugin should be in issue_importer_xls folder, otherwise it will not work.

**Comment by lajish** [comment]
Click action over 'Generate Excel File' does nothin.....
I am using redmine 2.5.2
Here goes production log content

Started GET "/redmine/settings/plugin/issue_importer_xls" for 127.0.0.1 at 2014-07-23 18:36:39 +0530
Processing by SettingsController#plugin as HTML
  Parameters: {"id"=>"issue_importer_xls"}
  Current user: lajish.l@abcd.com (id=4)
  Rendered plugins/issue_importer_xls/app/views/settings/_issue_importer_setting.html.erb (52.7ms)
  Rendered settings/plugin.html.erb within layouts/admin (85.9ms)
  Rendered admin/_menu.html.erb (10.7ms)
  Rendered plugins/sidebar_hide/app/views/sidebar/_hideButton_partial.html.erb (2.0ms)
  Rendered layouts/base.html.erb (97.7ms)
Completed 200 OK in 314.5ms (Views: 202.2ms | ActiveRecord: 53.7ms)

There is no log after this.....

**Comment by guruprasad1** [comment]
Can you please post browser request logs from firebug or any other way possible.
After clicking on Generate Excel Sheet it will generate Xls file .
Logs from production 

Started POST "/generate_excel_sheet" for 127.0.0.1 at 2014-07-23 18:59:47 +0530
Processing by ExcelSheetController#generate_excel_sheet as _/_
  Parameters: {"task_column"=>"0", "task_description_column"=>"1", "average_hour_column"=>"4", "start_date_column"=>"2", "end_date_column"=>"3", "asignee_name_column"=>"5"}
  Current user: guru.zapate (id=8)
  Rendered text template (0.0ms)
Completed 200 OK in 161.2ms (Views: 34.8ms | ActiveRecord: 0.7ms)
Started GET "/settings/plugin/public/uploads/exports/Redmine_Sample_Issue_Sheet.xls" for 127.0.0.1 at 2014-07-23 18:59:47 +0530
Processing by ExcelSheetController#export_excel_sheet as HTML
  Current user: guru.zapate (id=8)
Sent file public/uploads/exports/Redmine_Sample_Issue_Sheet.xls (0.1ms)
Completed 200 OK in 6.4ms (ActiveRecord: 0.7ms)

I doubt that redmine application is not receiving request for generating xls sheet.

**Comment by lajish** [comment]
I also tried it in firefox but i dont know to look into this issue using firebug. Please provide guidance for same.

**Comment by justin-castleman** [comment]
I am also having issues getting this to work. When I try to go in and configure the plugin I get the same error as the post above.

http://monosnap.com/image/QOU0UEQkAeDbrRZ6SFyv3AwZJ8eJcF

I did what was suggested and changed folder name in the plugin directory to the following

issue_importer_xls

and I am still getting the error.

**Comment by jwymanm** [comment]
The error looks to be because public/uploads/exports needs to be created at your top level redmine directory. Once that is created it seems to work fine.

**Comment by guruprasad1** [comment]
@jwymanm  Yes its true.If excel file is requested first time then  it will try to create export folder inside public->uploads.

@lajish  & @justin-castleman  : Can you make sure that Redmine Application has permission to create folder at {Redmine-Root}/public/uploads.And by which user you guys start Redmine Application?

**Comment by lajish** [comment]
Hi there,
I have 'Full Control' rights in that folder. Please have a look over picture pasted below

![image](https://cloud.githubusercontent.com/assets/6768694/3684181/79b65450-12f6-11e4-8ccb-d75188e45b44.png)

**Comment by guruprasad1** [comment]
@lajish  : It seems there are no permission issues present on your  machine. If you use firefox then hit F12 and go to console tab and then request for Excel template.By this way you will be able to see the request going through browser.I have attached the picture to  you give an idea about how it looks.
![screenshot from 2014-07-24 13 53 53](https://cloud.githubusercontent.com/assets/5365218/3685507/2cdc9e4e-130c-11e4-9d40-81794e9ae843.png)

**Comment by lajish** [comment]
HI there,
Hmm...Above screenshot reveals that you guys are using this plugin from Linux Mint(Don't which version of firefox)
I tried same thing in windows(Firefox 30.0) and it is showing Error 404 not found. I think you will get same issue if your firefox version is same as that of mine. Please look into image which i pasted below

![image](https://cloud.githubusercontent.com/assets/6768694/3685693/a5680978-130e-11e4-896b-a7ab54a15330.png)

**Comment by guruprasad1** [comment]
@lajish  From above picture I am assuming you are running Redmine Application on port 8080 and  as a root application directly not as another application context such as 127.0.0.1/redmine/login 
Right?

**Comment by lajish** [comment]
Yes, you are right. As i have another application which uses port default port (i.e.80). To avoid any issues related to usage of default port, I changed port configuration to 8080.

**Comment by guruprasad1** [comment]
Checked the routes file and action mapping for 'generate_excel_sheet' which seems ok .
I tried to generate  this issue on my local machine via different alternatives but not able to do so.
Make sure that plugin folder has 'issue_importer_xls' name.
Are you sure that there are no logs after below mentioned log trace

Started GET "/redmine/settings/plugin/issue_importer_xls" for 127.0.0.1 at 2014-07-23 18:36:39 +0530
Processing by SettingsController#plugin as HTML
Parameters: {"id"=>"issue_importer_xls"}
Current user: lajish.l@abcd.com (id=4)
Rendered plugins/issue_importer_xls/app/views/settings/_issue_importer_setting.html.erb (52.7ms)
Rendered settings/plugin.html.erb within layouts/admin (85.9ms)
Rendered admin/_menu.html.erb (10.7ms)
Rendered plugins/sidebar_hide/app/views/sidebar/_hideButton_partial.html.erb (2.0ms)
Rendered layouts/base.html.erb (97.7ms)
Completed 200 OK in 314.5ms (Views: 202.2ms | ActiveRecord: 53.7ms)

If any log is available then let me know.

**Comment by vrstud** [comment]
Hello:
I am having the same problem and I can't create a template or import.
I am currently running redmine from Bitnami 3.0.3-0 on Windows.
I had to fix a small bug since I am running ruby 4.0 in the routes.rb file.  The "match" function is causing an error.  I had to add ", via: :all" to the end of the line in order for it to compile. 
Once I did that, I can get the plugin to load but I can't get it to generate sample excel spreadsheet.  I also am getting an error on importing but I figure this is because the excel template I generated wasn't formatted properly.  

Any help would be appreciated.
